src/bootstrap/src/utils/proc_macro_deps.rs

```diff
@@ -0,0 +1,155 @@
+/// Do not update manually - use `./x.py test tidy --bless`
+/// Holds all direct and indirect dependencies of proc-macro crates in tree.
+/// See https://github.com/rust-lang/rust/issues/134863
+pub static CRATES: &[&str] = &[
+    // tidy-alphabetical-start
+    "annotate-snippets",
+    "anstyle",
+    "autocfg",
+    "basic-toml",
+    "bitflags",
+    "block-buffer",
+    "bumpalo",
+    "byteorder",
+    "cfg-if",
+    "clap_derive",
+    "color-print-proc-macro",
+    "cpufeatures",
+    "crossbeam-channel",
+    "crossbeam-deque",
+    "crossbeam-epoch",
+    "crossbeam-utils",
+    "crypto-common",
+    "darling",
+    "darling_core",
+    "darling_macro",
+    "derive-where",
+    "derive_builder_core",
+    "derive_setters",
+    "digest",
+    "displaydoc",
+    "either",
+    "equivalent",
+    "fluent-bundle",
+    "fluent-langneg",
+    "fluent-syntax",
+    "fnv",
+    "foldhash",
+    "futf",
+    "futures-macro",
+    "generic-array",
+    "getrandom",
+    "hashbrown",
+    "heck",
+    "hermit-abi",
+    "html5ever",
+    "icu_provider_macros",
+    "ident_case",
+    "indexmap",
+    "intl-memoizer",
+    "intl_pluralrules",
+    "itoa",
+    "libc",
+    "lock_api",
+    "log",
+    "mac",
+    "markup5ever",
+    "memchr",
+    "mime",
+    "mime_guess",
+    "minimal-lexical",
+    "new_debug_unreachable",
+    "nom",
+    "num_cpus",
+    "once_cell",
+    "parking_lot",
+    "parking_lot_core",
+    "pest",
+    "pest_generator",
+    "pest_meta",
+    "phf",
+    "phf_codegen",
+    "phf_generator",
+    "phf_shared",
+    "ppv-lite86",
+    "precomputed-hash",
+    "proc-macro-hack",
+    "proc-macro2",
+    "quote",
+    "rand",
+    "rand_chacha",
+    "rand_core",
+    "rayon",
+    "rayon-core",
+    "redox_syscall",
+    "rinja_derive",
+    "rinja_parser",
+    "rustc-hash",
+    "rustc-rayon",
+    "rustc-rayon-core",
+    "rustc_fluent_macro",
```

Comment:
This list seems to include the proc-macros themself too, which is not really necessary as bin/rustc.rs already checks for `--crate-type proc-macro`. If it isn't too much effort, removing them from the list may slightly reduce the amount of times this list needs to be updated.